optax/_src/loss.py

```diff
@@ -503,3 +503,19 @@ def kl_divergence_with_log_targets(log_predictions: chex.Array,
   chex.assert_type([log_predictions, log_targets], float)
   loss = jnp.exp(log_targets) * (log_targets - log_predictions)
   return jnp.sum(loss, axis=-1)
+
+def hinge_loss(predictions: chex.Array, targets: chex.Array) -> chex.Array:
+  """
+  Computes the hinge loss for Binary Classification.
+
+  Args:
+    predictions: Predicted values with shape [..., dim].
+    targets: Target values with shape [..., dim].
+      Expected to be in range [-1, 1].
+
+  Returns:
+    Average Hinge Loss
+  """
+  assert predictions.shape[0] == targets.shape[0], "Predictions and Target lengths must be consistent."
+  # Binary Case
+  return jnp.mean(jnp.maximum(0, 1 - predictions * targets))
```

Comment:
As I wrote above, I don't think we should add the `jnp.mean` in analogy to other optimizer (see #193) and also since this would take the mean over the whole array - but the user might want some axis to be excluded.